components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -153,6 +158,30 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
     this.isDestroyed.complete();
   }
 
+  public patchResourceValue(form, event) {
+    form.controls['type'].setValue(event.target.value);
+    this.attributeList = this.attributes[form.get('type').value.toLowerCase()];
+    this.selectResourceChefOption = false;
+  }
+
+  public patchAttributeValue(form, event, i) {
+    this.currentRuleIndex = i;
+    form.controls.conditions.controls[i].controls['attribute'].setValue(event.target.value);
+    this.selectAttributeChefOption = false;
+    if (this.editingRule) {
+      form.controls.conditions.controls[i].controls['attribute'].markAsDirty();
+    }
```

Comment:
This causes a bug.
Previously, for an existing rule, if you start with the attribute at value "A" and change to "B", that would enable the "Save Rule" button. But further, if you change it back to "A", that would disable the "Save Rule" button. That dirty/clean check is broken here due to this block, but might also be tied to switching from `<select>` to `<chef-select>`. 

So first see if you can determine if the bug is due to using `<chef-select>` (by changing it back to `<select>` and deleting L171-173). If that is the case, then try to fix `<chef-select>`--we do not want custom code added in _this_ file to do it, because then everytime we use `<chef-select>` we have to duplicate that kludge.

Reply:
@msorens thanks for review, I have verified this bug is due to using ` <chef-select> ` so fixing it in ` chef-ui-library ` it self so we can avoid code duplication